packages/components/src/slot-fill/slot.tsx

```diff
@@ -43,11 +50,11 @@ class SlotComponent extends Component {
 		unregisterSlot( this.props.name, this );
 	}
 
-	componentDidUpdate( prevProps ) {
+	componentDidUpdate( prevProps: BaseSlotComponentProps ) {
 		const { name, unregisterSlot, registerSlot } = this.props;
 
 		if ( prevProps.name !== name ) {
-			unregisterSlot( prevProps.name );
+			unregisterSlot( prevProps.name, this );
```

Comment:
Small bug fix. 

If I could find a way to use something other than `this`, I could change it to a function component.

Reply:
This may deserve its own PR TBH 😉

Reply:
Agreed with Marin, we should extract this fix to a separate PR

Reply:
@ciampo 

https://github.com/WordPress/gutenberg/pull/54765

Pull Request created.